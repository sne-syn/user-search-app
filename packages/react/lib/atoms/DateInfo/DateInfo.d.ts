import { FC } from "react";
interface DateInfoProps {
    date: string | null;
}
declare const DateInfo: FC<DateInfoProps>;
export default DateInfo;
