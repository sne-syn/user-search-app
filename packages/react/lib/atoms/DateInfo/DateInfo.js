import React from 'react';

const getHumanReadableDate = (date) => {
    const dateFormatted = new Date(date);
    const day = dateFormatted.getDay();
    const month = dateFormatted.toLocaleString("default", { month: "short" });
    const year = dateFormatted.getFullYear();
    return `${day} ${month} ${year}`;
};
const DateInfo = ({ date }) => {
    return (React.createElement("div", { className: "dse-date" },
        React.createElement("span", null, "Joined "),
        date && React.createElement("time", { dateTime: date }, getHumanReadableDate(date))));
};

export { DateInfo as default };
//# sourceMappingURL=DateInfo.js.map
