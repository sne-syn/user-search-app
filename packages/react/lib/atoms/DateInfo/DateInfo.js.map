{"version":3,"file":"DateInfo.js","sources":[],"sourcesContent":[],"names":[],"mappings":";;;;;;;;;;;;;;;;;"}