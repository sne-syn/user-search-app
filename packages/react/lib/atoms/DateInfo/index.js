export { default } from './DateInfo.js';
//# sourceMappingURL=index.js.map
