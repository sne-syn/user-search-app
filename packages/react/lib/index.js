export { default as Avatar } from './atoms/Avatar/Avatar.js';
export { default as Button } from './atoms/Button/Button.js';
export { default as CardSubtitle } from './atoms/CardSubtitle/CardSubtitle.js';
export { default as CardTitle } from './atoms/CardTitle/CardTitle.js';
export { default as Icon } from './atoms/Icon/Icon.js';
export { default as Input } from './atoms/Input/Input.js';
export { default as Logo } from './atoms/Logo/Logo.js';
export { default as DateInfo } from './atoms/DateInfo/DateInfo.js';
export { default as DefinitionList } from './atoms/DefinitionList/DefinitionList.js';
export { default as ErrorMessage } from './atoms/ErrorMessage/ErrorMessage.js';
export { default as TextBlock } from './atoms/TextBlock/TextBlock.js';
export { default as ThemeToggler } from './atoms/ThemeToggler/ThemeToggler.js';
export { default as VisuallyHidden } from './atoms/VisuallyHidden/VisuallyHidden.js';
export { default as Wrapper } from './atoms/Wrapper/Wrapper.js';
//# sourceMappingURL=index.js.map
