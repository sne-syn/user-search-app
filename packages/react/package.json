{
  "name": "@user-search-app/react",
  "version": "1.0.0",
  "main": "lib/index.js",
  "license": "MIT",
  "scripts": {
    "build": "rollup -c",
    "dev": "yarn build --watch",
    "test": "jest --verbose",
    "test:watch": "yarn test --watch"
  },
  "devDependencies": {
    "@babel/preset-env": "^7.18.10",
    "@babel/preset-react": "^7.18.6",
    "@babel/preset-typescript": "^7.18.6",
    "@rollup/plugin-image": "^2.1.1",
    "@testing-library/jest-dom": "^5.16.5",
    "@testing-library/react": "^13.3.0",
    "@types/jest": "^28.1.6",
    "@types/react": "^18.0.15",
    "jest": "^28.1.3",
    "jest-environment-jsdom": "^28.1.3",
    "react": "^18.2.0",
    "rollup": "^2.77.0",
    "rollup-plugin-img": "^1.1.0",
    "rollup-plugin-typescript2": "^0.32.1",
    "typescript": "^4.7.4"
  }
}
