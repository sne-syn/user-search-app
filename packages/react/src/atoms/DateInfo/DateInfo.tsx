import React, { FC } from "react";

interface DateInfoProps {
  date: string | null;
}
const getHumanReadableDate = (date: string): string => {
  const dateFormatted = new Date(date);

  const day = dateFormatted.getDay();
  const month = dateFormatted.toLocaleString("default", { month: "short" });
  const year = dateFormatted.getFullYear();

  return `${day} ${month} ${year}`;
};

const DateInfo: FC<DateInfoProps> = ({ date }) => {
  return (
    <>
      {date && (
        <div className="dse-date" data-testid="date-info">
          <span>Joined </span>
          <time dateTime={date}>{getHumanReadableDate(date)}</time>
        </div>
      )}
    </>
  );
};

export default DateInfo;
