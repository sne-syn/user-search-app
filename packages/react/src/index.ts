import Avatar from "./atoms/Avatar";
import Button from "./atoms/Button";
import CardSubtitle from "./atoms/CardSubtitle";
import CardTitle from "./atoms/CardTitle";
import Input from "./atoms/Input";
import ListContent from "./atoms/ListContent";
import ListIcon from "./atoms/ListIcon";
import Logo from "./atoms/Logo";
import DateInfo from "./atoms/DateInfo";
import DefinitionList from "./atoms/DefinitionList";
import ErrorMessage from "./atoms/ErrorMessage";
import TextBlock from "./atoms/TextBlock";
import ThemeToggler from "./atoms/ThemeToggler";
import VisuallyHidden from "./atoms/VisuallyHidden";
import Wrapper from "./atoms/Wrapper";

export {
  Avatar,
  Button,
  CardSubtitle,
  CardTitle,
  Input,
  ListContent,
  ListIcon,
  Logo,
  DateInfo,
  DefinitionList,
  ErrorMessage,
  TextBlock,
  ThemeToggler,
  VisuallyHidden,
  Wrapper,
};
