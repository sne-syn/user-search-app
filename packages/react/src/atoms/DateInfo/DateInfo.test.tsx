import React from "react";
import DateInfo from "./DateInfo";
import { render, screen } from "@testing-library/react";

const renderComponent = (date: string | null) => {
  return render(<DateInfo date={date} />);
};
describe("DateInfo", () => {
  test("renders with date on human readable format", () => {
    renderComponent("2008-01-14T04:33:35Z");
    expect(screen.getByTestId("date-info")).toBeInTheDocument();
    expect(screen.getByTestId("date-info")).toHaveTextContent("1 Jan 2008");
    expect(screen.getByTestId("date-info")).toMatchSnapshot();
  });
  test("doesn't render if the date is absent", () => {
    renderComponent("");
    expect(screen.queryByTestId("date-info")).not.toBeInTheDocument();
    expect(screen.queryByTestId("date-info")).toMatchSnapshot();
  });
});
