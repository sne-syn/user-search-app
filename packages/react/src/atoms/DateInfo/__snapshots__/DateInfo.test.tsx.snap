// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateInfo doesn't render if the date is absent 1`] = `null`;

exports[`DateInfo renders with date on human readable format 1`] = `
<div
  class="dse-date"
  data-testid="date-info"
>
  <span>
    Joined 
  </span>
  <time
    datetime="2008-01-14T04:33:35Z"
  >
    1 Jan 2008
  </time>
</div>
`;
